pkg/s3/client.go

```diff
@@ -94,17 +94,35 @@ func (client *s3Client) CreatePrefix(bucketName string, prefix string) error {
 }
 
 func (client *s3Client) RemovePrefix(bucketName string, prefix string) error {
-	if err := client.removeObjects(bucketName, prefix); err != nil {
-		return err
+	var err error
+
+	if err = client.removeObjects(bucketName, prefix); err == nil {
+		return client.minio.RemoveObject(client.ctx, bucketName, prefix, minio.RemoveObjectOptions{})
 	}
-	return client.minio.RemoveObject(client.ctx, bucketName, prefix, minio.RemoveObjectOptions{})
+
+	glog.Warning("removeObjects failed with: ", err, ", will try removeObjectsOneByOne")
```

Comment:
Can you explain in which cases the `removeObjects` can fail and maybe also add that to a comment?

Also a small nit, you can use `glog.Warningf` to format the string.

Reply:
Digitalocean not support file lock, in remove objects have set GovernanceBypass header , but digitalocean not support this header.